src/main/java/com/cargohub/security/jwt/JwtTokenProvider.java

```diff
@@ -0,0 +1,106 @@
+package com.cargohub.security.jwt;
+
+import com.cargohub.entities.RoleEntity;
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jws;
+import io.jsonwebtoken.JwtException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Component;
+import javax.annotation.PostConstruct;
+import javax.servlet.http.HttpServletRequest;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.Date;
+import java.util.List;
+
+
+@Component
+public class JwtTokenProvider {
+
+    @Value("${jwt.token.secret}")
+    private String secret;
+
+    @Value("${jwt.token.expired}")
+    private long validityInMilliseconds;
+
+
+    @Autowired
+    private UserDetailsService userDetailsService;
+
+    @Bean
+    public BCryptPasswordEncoder passwordEncoder() {
+        BCryptPasswordEncoder bCryptPasswordEncoder = new BCryptPasswordEncoder();
+        return bCryptPasswordEncoder;
```

Comment:
Why it removed from Config class?

Reply:
This bean moved to JwtTokenProvider class during the bug fixing. If it's necessary this bean can be moved back to Config.